plumbing/transport/internal/common/common.go

```diff
@@ -149,10 +149,29 @@ func (c *client) listenFirstError(r io.Reader) chan string {
 	errLine := make(chan string, 1)
 	go func() {
 		s := bufio.NewScanner(r)
-		if s.Scan() {
-			errLine <- s.Text()
-		} else {
-			close(errLine)
+		var firstLine string
+		for {
+			if s.Scan() {
+				line := s.Text()
+				if len(firstLine) == 0 {
+					// store the first line as a fallback in case
+					// we can't find a line indicating the actual
+					// error
+					firstLine = line
+				}
+				idx := strings.Index(line, "ERROR: ")
+				if idx != -1 {
+					errLine <- line[idx:]
+					break
+				}
```

Comment:
We could simplify this by having a regex of line patterns we want to ignore (e.g. `^remote: (=)*(\r\n|\r|\n){0,1}$`), that way we don't need to hold anything in memory while processing the other lines.

Reply:
Even if we used the skip approach using a regex rather than the match approach as implemented here we would have to hold the first line in memory, anyway so that we can fall back to returning that which might still be better than having nothing to return in case all lines match the regex to be skipped. wdyt?

Reply:
If the regex is very specific, it should discard only the unhelpful lines - which we would never want to return anyway. In that case, we could return the first valid line, or fallback to a generic error message.

Reply:
Yeah I agree that returning `remote: ` doesn't really help. I was just trying to be cognizant of backwards-compatability but I suppose no business logic should actually depend on the text of an error. 😊 I'll adapt the PR accordingly.

Reply:
I pushed the changes as agreed. ptal. 🙏🏻